### PR TITLE
feat(ci): implement release detection from README roadmap

### LINKED ISSUE #100: [release-management] Detect current release from README roadmap
## Context

The changelog generation workflow needs a reliable way to determine which release is currently in progress.

In this repository, the source of truth for release planning is the "Releases & Changelog" section in `README.md`, where the release table contains version, name, planned date, status, and details.

The current release is defined as the row whose `Status` is `In Progress` and whose `Details` is `Plan`.

Without a deterministic way to identify the active release from the README, the automation cannot know which milestone and set of tasks must be considered for changelog generation.

## Objective

Implement logic to detect the current release from the `README.md` release roadmap table and extract its release metadata, especially the release version and release name.

### Included
- Read the `README.md` file
- Locate the "Releases & Changelog" section
- Parse the release roadmap table
- Detect the row representing the current release
- Extract at least:
  - release version
  - release name
  - planned date
  - status
  - details
- Return structured release metadata for downstream CI steps

### Excluded
- Querying GitHub Issues or Milestones
- Generating the changelog entry itself
- Updating `CHANGELOG.md`
- CI workflow triggering logic

### Scenario 1 — Happy Path
**Given**  
the `README.md` contains a valid "Releases & Changelog" table  
and one row is marked as `🚧 In Progress` with `Plan` in the details column  

**When**  
the release detection logic runs  

**Then**  
the current release is identified correctly  
and its version and name are returned as structured metadata  

### Scenario 2 — Edge Case
**Given**  
the release roadmap table contains released and planned versions in addition to the current one  

**When**  
the release detection logic runs  

**Then**  
only the row marked as the current release is selected  
and other rows are ignored  

### Scenario 3 — Failure Case
**Given**  
the `README.md` does not contain a valid release roadmap table  
or no row matches the expected current release criteria  

**When**  
the release detection logic runs  

**Then**  
a clear and specific error is raised  
and the workflow does not continue with ambiguous release metadata  

## Acceptance Criteria

- [x] The current release is derived from the `README.md` release roadmap table
- [x] The release version and release name are extracted correctly
- [x] The logic fails explicitly when the current release cannot be determined
- [x] Output is structured and reusable by downstream changelog generation steps
- [x] Unit tests cover happy path, edge case, and failure case scenarios

## Notes (Optional)

The current release is the row in the release roadmap table with:
- `Status = 🚧 In Progress`
- `Details = Plan`

The release name extracted here will be used later to match GitHub Milestones.

## Related Issue

Closes #

### PR DESCRIPTION
**Pull request by brunolnetto**
## Changes

- Add `tools/changelog_generator/__init__.py` — makes the package importable
- Add `tools/changelog_generator/release_detector.py` — `ReleaseDetector` reads `README.md`, locates the Release Roadmap table, and returns a `ReleaseMetadata` dataclass for the first row whose status is `🚧 In Progress` and whose details column contains a `[Plan](url)` link
- Add `tools/changelog_generator/tests/conftest.py` — adds `tools/` to `sys.path` so all tests can import `changelog_generator.*`
- Add `tools/changelog_generator/tests/test_release_detector.py` — 7 unit tests covering the happy path, multiple In-Progress rows edge case, and all failure paths

## Motivation

Automated changelog generation (#103) requires knowing which release is currently in progress before querying GitHub for its tasks. This PR implements the first step of that pipeline: detecting the current release from the project's README roadmap table.

## Impact

- No changes to existing source code or tests
- New `tools/changelog_generator/` package (Python, stdlib-only)
- No new runtime dependencies

## Checklist

- [x] Tests added for new code
- [x] All new code uses stdlib only (no external dependencies)
- [x] Commit message follows the convention

## Related Issue

Closes #100

### CODE REVIEW

File: tests/unit/changelog_generator/test_release_detector.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Releases & Roadmap
+
+Release Roadmap:
+
+| Version | Name | Status | Planned Date | Details |
+|--------|------|--------|---------|---------|
+| v0.2.1 | Old Release | ✅ Released | 2026-03-01 | [Notes](https://example.com/notes) |
+| v0.2.2 | Curriculum Graph API Foundation | 🚧 In Progress | 2026-06-15 | [Plan](https://example.com/186) |
+| v0.3.0 | Orchestrator Core | 🔜 Planned | 2026-09-01 | [Plan](https://example.com/185) |
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tests/unit/changelog_generator/test_release_detector.py:26]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/223#discussion_r3267050341)

File: tools/changelog_generator/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import sys
+from pathlib import Path
+
+# Add tools/ to sys.path so that `changelog_generator` is importable as a package.
+sys.path.insert(0, str(Path(__file__).parent.parent.parent))
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/tests/conftest.py]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/223#discussion_r3267050409)

**Review by OLeandroRodrigues — APPROVED** [review]
